doc: ESM documentation consolidation and reordering (#36046)
nodejs/node: doc/api/process.md

* `filename` {string}
* `flags` {os.constants.dlopen} **Default:** `os.constants.dlopen.RTLD_LAZY`

The `process.dlopen()` method allows dynamically loading shared objects. It is
primarily used by `require()` to load C++ Addons, and should not be used
directly, except in special cases. In other words, [`require()`][] should be
preferred over `process.dlopen()` unless there are specific reasons such as
custom dlopen flags or loading from ES modules.

The `flags` argument is an integer that allows to specify dlopen
behavior. See the [`os.constants.dlopen`][] documentation for details.

An important requirement when calling `process.dlopen()` is that the `module`

> instance must be passed. Functions exported by the C++ Addon will then be

Reviewer: ```suggestion
instance must be passed. Functions exported by the C++ Addon are then
```

accessible via `module.exports`.

The example below shows how to load a C++ Addon, named `local.node`,

> that exports a `foo` function. All the symbols will be loaded before

Reviewer: ```suggestion
that exports a `foo` function. All the symbols are loaded before
```
